multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/PollServiceBrokersOperationsExecution.java

```diff
@@ -0,0 +1,52 @@
+package org.cloudfoundry.multiapps.controller.process.steps;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+
+public class PollServiceBrokersOperationsExecution extends PollServiceBrokerOperationsExecution {
+
+    @Override
+    public AsyncExecutionState execute(ProcessContext context) {
+        Map<String, String> serviceBrokerNamesJobIds = context.getVariable(Variables.SERVICE_BROKER_NAMES_JOB_IDS);
+
+        Map<String, AsyncExecutionState> serviceBrokerNamesExecutionStates = new HashMap<>();
+        for (Entry<String, String> serviceBrokerNameJobId : serviceBrokerNamesJobIds.entrySet()) {
+            String serviceBrokerName = serviceBrokerNameJobId.getKey();
+            String jobId = serviceBrokerNameJobId.getValue();
+
+            AsyncExecutionState asyncJobStatus = checkAsyncJobStatus(context, serviceBrokerName, jobId);
+            serviceBrokerNamesExecutionStates.put(serviceBrokerName, asyncJobStatus);
+        }
+
+        if (serviceBrokerNamesExecutionStates.entrySet()
+                                             .stream()
+                                             .anyMatch(serviceBrokerNameExecutionState -> serviceBrokerNameExecutionState.getValue() == AsyncExecutionState.ERROR)) {
+            return AsyncExecutionState.ERROR;
+        }
+
+        if (serviceBrokerNamesExecutionStates.entrySet()
+                                             .stream()
+                                             .allMatch(serviceBrokerNameExecutionState -> serviceBrokerNameExecutionState.getValue() == AsyncExecutionState.FINISHED)) {
+            return AsyncExecutionState.FINISHED;
+        }
+
+        Map<String, String> updatedServiceBrokerNamesJobIds = new HashMap<>(serviceBrokerNamesJobIds);
+        serviceBrokerNamesExecutionStates.entrySet()
+                                         .stream()
+                                         .filter(serviceBrokerNameExecutionState -> serviceBrokerNameExecutionState.getValue() == AsyncExecutionState.FINISHED)
+                                         .forEach(serviceBrokerNameExecutionState -> updatedServiceBrokerNamesJobIds.remove(serviceBrokerNameExecutionState.getKey()));
+
```

Comment:
I think this could be simplified like so:
```java
Map<String, String> serviceBrokerJobsStillInProgress = new HashMap<>();
for (var entry : serviceBrokerNamesJobIds.entrySet()) {
   String serviceBrokerName = entry .getKey();
   String jobId = entry .getValue();
   
   AsyncExecutionState jobState = checkAsyncJobStatus(context, serviceBrokerName, jobId);
   if (jobState == AsyncExecutionState.ERROR) {
      return AsyncExecutionState.ERROR;
   }
   if (jobState != AsyncExecutionState.FINISHED) {
      serviceBrokerJobsStillInProgress.put(serviceBrokerName, jobId);
   }
}

if (serviceBrokerJobsStillInProgress.isEmpty()) {
   return AsyncExecutionState.FINISHED;
}

context.set(SERVICE_BROKER_NAMES_JOB_IDS, serviceBrokerJobsStillInProgress);
```